buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -429,7 +429,7 @@ class BeamModulePlugin implements Plugin<Project> {
   }
 
   def isRelease(Project project) {
-    return parseBooleanProperty(project, 'isRelease');
+    return !project.version.endsWith('-SNAPSHOT')
```

Comment:
Doesn't this remove our ability to run non-release jobs against the RC commit before we're ready to publish the release?

Reply:
I don't actually understand the question. Can you restate it somehow? I don't understand what these things are:

 - "non-release jobs"
 - "against the RC commit"

Reply:
"the RC commit" - the commit that we tag for a release candidate. E.g. https://github.com/apache/beam/tree/v2.51.0-RC1

"non-release jobs" - any job that was previously expecting the version to end in `-SNAPSHOT` or has other expectations depending on the `isRelease` variable. This expectation is baked in all over the place FWIW - https://github.com/search?q=repo%3Aapache%2Fbeam+isRelease+language%3AGradle&type=code&l=Gradle

Reply:
Oh yes, I'm aware of it being baked in all over the place, that's why I left it. Builds at the RC commit should always have `-PisRelease` or they won't do what we need to do with the RC.

If we have a job that expects the version to end in `-SNAPSHOT` that job is probably broken.

Reply:
I do need to fix the ones that are directly checking `hasProperty` instead of calling the method though.

Eventually we should be eliminating this concept, which should probably not exist.

Reply:
Converted to assigning a boolean, deliberately causing any dependencies on the prior function to break so we can catch them in review.